examples/apps/ncp/main.c

```diff
@@ -97,3 +97,24 @@ int main(int argc, char *argv[])
 
     return 0;
 }
+
+
+/*
+ * Provide, if required an "otPlatLog()" function
+ */
+
+#if ((OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_NONE) ||  (OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_APP_STREAM))
```

Comment:
If we make the default to be `PLATFORM_DEFINED`, I don't think we need to add stubs in each platform. Each example platform has its own `openthread-core-<radio/platform>-config.h` file. I think we can just define `OPENTHREAD_CONFIG_LOG_OUTPUT` in these files (to `NONE` or whatever makes sense for the example platform).